docs/execution-checkpoints.md

```diff
@@ -0,0 +1,5 @@
+# Checkpoints
+
+This is a blockchain asset/contract deployment tool.
+Since blockchain data is immutable (can't be changed) there is no way to rewrite its contents and attempts will result in an error.
+To prevent this we propose a checkpont-based model for versioning the contents of the Algorand blockchain.
```

Comment:
Please add more information her:

+ how user is using it, 
+ how he can add metadata using deployer
+ how he can access data in other scripts.
+ when the checkpoint is saved

Reply:
Updated. Please read again (not here, in the changed files)